### PR TITLE
ISDB-268

### PR DESCRIPTION
**Pull request by parvezhmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ISDB-268

### Change description ###

- C51 form related changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by hmcts-kodatala — COMMENTED** [review]
Pls check all review comments and close it

File: src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC51ServiceTest.java [hunk 1]
```diff
@@ -45,40 +62,49 @@ void testC51MandatoryErrorWhileDoingValidation() {
...
     }
 
     @Test
-    void testC51OptionalFieldsWarningsWhileDoingValidation() {
-        BulkScanValidationRequest bulkScanValidationRequest = BulkScanValidationRequest.builder().ocrdatafields(
-                TestDataUtil.getC51ErrorData()).build();
-        BulkScanValidationResponse res = bulkScanValidationService.validate(bulkScanValidationRequest);
-        assertTrue(res.getWarnings().items.contains(String.format(DATE_FORMAT_MESSAGE, "applicant2_dateOfBirth")));
-    }
+    void testTransform() throws IOException, JSONException {
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC51ServiceTest.java:89]: it's already there.

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -36,6 +36,7 @@ spring:
       - c2-form
       - a1-form
       - a58-transform
+      - c51-transform
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply src/main/resources/application.yaml:40]: done

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/controllers/BulkScanC51FunctionalTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                .body(bulkScanTransformRequest)
+                .when()
+                .contentType("application/json")
+                .post("/transform-exception-record");
+
+        response.then().assertThat().statusCode(HttpStatus.OK.value());
+
+        JSONAssert.assertEquals(bulkScanTransformResponse, response.getBody().asString(), true);
+    }
+}
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/controllers/BulkScanC51FunctionalTest.java:146]: Added.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC51Service.java [hunk 4]
```diff
@@ -27,14 +52,95 @@ public FormType getCaseType() {
...
+
+        if (!validField) {
+
+            bulkScanValidationResponse.setStatus(Status.ERRORS);
+
+            List<String> items = bulkScanValidationResponse.getErrors()
+                    .getItems();
+
+            items.add(String.format(BulkScanConstants.XOR_CONDITIONAL_FIELDS_MESSAGE, conditionalFields));
+        }
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC51Service.java:117]: Done

**Review by hmcts-kodatala — APPROVED** [review]
OK

**Review by SangeethaPonnusamy — APPROVED** [review]

**Pull request merged**